HW1/P6/P6.py

```diff
@@ -0,0 +1,59 @@
+import numpy as np
+from functools import partial
+import findspark
+findspark.init()
+import pyspark
+
+
+def pre_process_data( filename, sc ):
+    wlist = sc.textFile( filename )
+    wlist_words = wlist.flatMap( lambda x: x.split() )
+    wlist_words_filter = wlist_words.filter( lambda x: not( x.isdigit() or x.isupper() or ( x[:-1].isupper() and x[-1:] == "." ) ) )
```

Comment:
According to the documentation on isupper, it will "Return true if all cased characters in the string are uppercase and there is at least one cased character, false otherwise." You want to check whether words containing only letters have all uppercase letters. This preprocessing removes things that it shouldn't. See https://piazza.com/class/icqfmi2aryn6yv?cid=226